src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -238,6 +240,63 @@ public void handleLogout() {
         showInnerParts();
     }
 
+    /**
+     * Sieves out the contacts based on the service type.
+     * @param resultMap Map of all related contacts.
+     * @param serviceType Type of service.
+     * @return List of Contacts of this service type.
+     */
+    private ObservableList<Contact> getfilteredServiceList(
+            Map<Contact, Collection<Service>> resultMap, String serviceType) {
+        List<Contact> serviceList = new ArrayList<>();
+        resultMap.forEach(((
+                contact, temp) -> resultMap.get(contact)
+                .forEach(service -> {
+                    if (service.getName().equals(serviceType)) {
+                        serviceList.add(contact);
+                    }
+                })));
```

Comment:
Consider using `Stream#reduce` in future.

Reply:
Will switch over if time permits thanks!